Dockerfile

```
@@ -85,6 +85,7 @@ RUN \
   rm -rf /var/lib/apt/lists/* && \
   c_rehash 2>&1
 
+COPY ./target/scripts/helpers/log.sh /usr/local/bin/helpers/log.sh
```

I think it does not really matter, if we copy all helpers at this stage. Also, this solution would not introduce another layer to the image.

```suggestion
COPY ./target/scripts/helpers /usr/local/bin/helpers
```

---

I rather we didn't... that was the whole point of making [this PR](https://github.com/docker-mailserver/docker-mailserver/pull/2197). If you're working on helpers, you have to rebuild the bulk of the image needlessly.

If we get around to a multi-stage build at some point, that'd be less of an issue. Getting all packages installed and ClamAV were both big time sinks to build time I think, plus ClamAV layer requires higher RAM (around 1-2GB IIRC) to build successfully, which usually meant paying twice as much for a VPS or configuring swap on disk. Not that the cost matters much if it's throwaway, and I've just been paying for instances I don't use monthly anyway instead of local VMs which I intend to finally get around to soon :sweat_smile:

---

@georglauterbach exactly, you explained why I only added `log.sh` here :D 👍

---

Ack.

> If we get around to a multi-stage build at some point,

I proposed that one time (squashing the image), but it was dismissed. IIRC it was because then it's not possible to inspect each layer on the final image anymore. So I do it only for my personal image build.

---

To be honest, I cannot remember :D Maybe having a multi-stage build with a final, squashed layer at the end is a nice thing to have. If we want to do it in the future, I'll be happy to review it :)

---

I think it's somewhere in the maintainers discussions. But without a search function, it's pretty hard to find it 😆

---

> IIRC it was because then it's not possible to inspect each layer on the final image anymore.

I think that's not true. We can compress directives/layers into groups via multi-stage build, you can copy each one to be a separate layer, not just a single build layer with a 2nd final image copying over the difference to single layer (_although that's a common approach_).

At least I'm pretty sure that is doable :)